### ISSUE TITLE
The documentation don't work well in accessibility mode

### ISSUE THREAD

**Issue by vedant-pandey** [leptos-rs/leptos#1327]
**Is your feature request related to a problem? Please describe.**
Using accessibillity mode features like Caret browsing on the documentation website is not currently feasible. This makes website less accessible individuals relying on browser aids

**Describe the solution you'd like**
Plan accessibility features to support and make sure current behaviour doesn't collide with Caret browsing.

**Describe alternatives you've considered**
Most regions cannot be reached without using a mouse since certain texts are not reachable by tab navigation and Caret browsing also doesn't work.

**Comment by gbj** [comment]
Could you please clarify which documentation website you're talking about? The book found at https://leptos-rs.github.io/leptos/ or the docs found at https://docs.rs/leptos? The book is built using mdbook. As far as I can tell it's possible to navigate it with the keyboard. I would be happy to try to address this, but I am not sure of the specific issues or what you mean by "not currently feasible," so more information would be useful.

**Comment by vedant-pandey** [comment]
Hi @gbj sorry for confusion this issue seems to be specific to [https://leptos-rs.github.io/leptos/](https://leptos-rs.github.io/leptos/), although navigation on [https://docs.rs/leptos](https://docs.rs/leptos) is also not very accessible, I'll try to give more details

For [https://leptos-rs.github.io/leptos/](https://leptos-rs.github.io/leptos/)
 - Press tab for navigation
 - There is no "Jump to content" and other accessibility navigations
 - After reaching the main content document, you cannot select the content on any specific line without using a mouse.
 - Turn on Caret browsing(F7 for most chromium based browsers)
 - Navigate up or down using arrow keys it will work as expected. If you use left of right arrow keys for navigation the page would change hindering the navigation

For docs.rs it mostly seems the accessibility navigation menu is just missing and hence you need to jump a lot of links before reaching the main content

For reference you can go to any google search result page, and click tab button it would open a hidden accessibility menu which would have options like jump to content, etc.

**Comment by gbj** [comment]
Thanks, this is helpful.

docs.rs docs are automatically generated from the doc comments included in a crate. We have no control over the accessibility of that website, and I'd think any accessibility issues would apply to the docs for every Rust crate. It's an official part of the Rust Project though, and they may be open to feedback re accessibility. The repo is here https://github.com/rust-lang/docs.rs

The book is generated with mdbook, which is also an official Rust project (https://github.com/rust-lang/mdBook). However in this case, we do control the website and are running mdbook ourselves to generate the pages, so there may be the possibility for more control over the accessibility menu etc — I believe mdbook has a plugin system. I'm pretty unfamiliar both of these (mdbook and accessibility) beyond the basics, so would appreciate help with this from anyone who's able to help.

**Comment by vedant-pandey** [comment]
Makes sense, I'll try to check out the docs and add any possible help I can

**Comment by gbj** [comment]
I am going to close this issue just because I'm cleaning up the issue tracker right now, and the issues here are all upstream in other projects (docs.rs and mdbook).
